MSAL/src/native_auth/public/state_machine/error/RetrieveAccessTokenError.swift

```diff
@@ -24,38 +24,39 @@
 
 import Foundation
 
-@objc
+@objcMembers
 public class RetrieveAccessTokenError: MSALNativeAuthError {
-    /// An error type indicating the type of error that occurred
-    @objc public let type: RetrieveAccessTokenErrorType
+    let type: RetrieveAccessTokenErrorType
 
     init(type: RetrieveAccessTokenErrorType, message: String? = nil) {
         self.type = type
-        super.init(message: message)
+        super.init(identifier: type.rawValue, message: message)
     }
 
+    /// Describes an error that provides messages describing why an error occurred and provides more information about the error.
     public override var errorDescription: String? {
-        if let description = super.errorDescription {
-            return description
-        }
-
-        switch type {
-        case .generalError:
-            return "General error"
-        case .refreshTokenExpired:
-            return "Refresh token expired"
-        case .tokenNotFound:
-            return "Token not found"
-        case .browserRequired:
-            return "Browser required"
-        }
+        return super.errorDescription ?? type.rawValue
+    }
+
+    /// Returns `true` if the error requires to use a browser.
+    public var isBrowserRequired: Bool {
+        return type == .browserRequired
+    }
+
+    /// Returns `true` if the refresh token has expired.
+    public var isRefreshTokenExpired: Bool {
+        return type == .refreshTokenExpired
+    }
+
+    /// Returns `true` if the existing token cannot be found.
+    public var isTokenNotFound: Bool {
+        return type == .tokenNotFound
     }
 }
 
-@objc
-public enum RetrieveAccessTokenErrorType: Int {
-    case generalError
-    case refreshTokenExpired
-    case tokenNotFound
-    case browserRequired
+public enum RetrieveAccessTokenErrorType: String, CaseIterable {
+    case browserRequired = "Browser required"
```

Comment:
Related to my previous comment, I think these string raw values for each case of the type should be a more fixed short computer-friendly string, e.g. `browser_required` that we know never ever to change. Having these strings as human-friendly short sentences is not as suitable for string comparison, using in `switch case` statements, etc.

Reply:
Indeed, we should have a distinction between `type` and `identifier`. Whichever is used for the switch should be computer friendly strings. Maybe have 2 enums one for type one for description/identifiers

Reply:
I think you are right, I've updated it. Check [this reply](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1923#discussion_r1401085666) to make it work with objc. I haven't tested it yet, but I think it could work.